sjsonnet/src-jvm/sjsonnet/SjsonnetMain.scala

```diff
@@ -159,41 +174,39 @@ object SjsonnetMain {
         }
       case (None, true) =>
         // YAML stream
+
         interp.interpret(os.read(path), OsPath(path)).flatMap {
           case arr: ujson.Arr =>
-            val renderedFiles: Seq[Either[String, String]] =
-              arr.value.toSeq.map{ v =>
-                Right(ujson.transform(v, new Renderer(indent = config.indent)).toString)
-              }
-            renderedFiles.collect{case Left(err) => err} match{
-              case Nil =>
-                val docs = renderedFiles.collect{case Right(yaml) => yaml}
-                val stream = if (docs.isEmpty) "" else docs.mkString("---\n", "\n---\n", "\n...")
-                Right[String, String](stream)
-              case errs =>
-                Left[String, String]("rendering errors:\n" + errs.mkString("\n"))
+            val stream = arr.value.toSeq match{
+              case Nil => ""
+              case Seq(single) =>
+                val renderer = rendererForConfig(new StringWriter(), config)
+                val suffix = if (isScalar(single)) "\n..." else ""
+                single.transform(renderer).toString + suffix
+              case multiple =>
+                multiple.zipWithIndex
+                  .map{
+                    case (v, i) =>
+                      val renderer = rendererForConfig(new StringWriter(), config)
+                      val rendered = v.transform(renderer).toString
+                      if (isScalar(v)) "--- " + rendered
+                      else if (i != 0) "---\n" + rendered
```

Comment:
This changes the streaming JSON style for scalar values to `"--- "`-prefixed rather than `"---\n"`-prefixed. While the old style was compatible, this new style is a lot more compact and more closely matches what other implementations like PyYAML do